lib/crypto/aes.toit

```diff
@@ -2,92 +2,155 @@
 // Use of this source code is governed by an MIT-style license that can be
 // found in the lib/LICENSE file.
 
+/**
+Base class for the hardware accelerated Advanced Encryption Standard (AES).
+
+See https://en.wikipedia.org/wiki/Advanced_Encryption_Standard.
+
+AES has multiple different modes it can use.
+The ones implemented here are CBC and ECB.
+*/
+abstract class Aes:
+  aes_ := ?
+
+  /**
+  Initialize an Aes class from a subclass. 
+  If the $initialization_vector is empty, then AES ECB mode is selected.
+  If the $initialization_vector has length 16, then AES CBC mode is selected.
+  */
+  constructor.from_subclass key/ByteArray initialization_vector/ByteArray encrypt/bool:
```

Comment:
Should this be module private? `constructor.initialize_`?

Reply:
It is now private